pkg/conversion/deep_equal.go

```diff
@@ -174,6 +174,9 @@ func (e Equalities) deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool,
 
 	switch v1.Kind() {
 	case reflect.Array:
+		if v1.Len() != v2.Len() {
+			return false
+		}
```

Comment:
Were you able to write a non-working test such that adding this line fixed it? Because this code is copied from the std lib and anything broken here is a bug in the std lib... but this seems to work fine: https://play.golang.org/p/koOwuqviR8